v2/options.go

```diff
@@ -100,14 +118,14 @@ func WithMultiplier(v float64) ExponentialOption {
 // value outside of this range is specified, the value will be silently
 // ignored and jittering is disabled.
 //
-// This option can be passed to ExponentialPolicy constructor
-func WithJitterFactor(v float64) ExponentialOption {
-	return &exponentialOption{option.New(identJitterFactor{}, v)}
+// This option can be passed to ExponentialPolicy or ConstantPolicy constructor
+func WithJitterFactor(v float64) JitterOption {
+	return &jitterOption{option.New(identJitterFactor{}, v)}
```

Comment:
`return &exponentialOption{constantOption{option.New(identJitterFactor)}}` should suffice

Reply:
```go
type CommonOption interface {
	ExponentialOption
	ConstantOption
}

func WithJitterFactor(v float64) CommonOption {
	return &exponentialOption{constantOption{option.New(identJitterFactor)}}
}
```

In this case, go compiler complains like the following, hmm:

```
Cannot use '&exponentialOption{constantOption{option.New(identJitterFactor)}}' (type *exponentialOption) as type CommonOption Type does not implement 'CommonOption' as some methods are missing: constantOption()
```

Reply:
Ah, oops. then I guess it's back to

```go
type CommonOption interface {
    ExponentialOption
    ConstantOption
}

type commonOption struct { Option }
func (*commonOption) exponentialOption() {}
func (*commonOption) constantOption() {}

func WithJitterFactor(v float64) CommonOption { 
  return &commonOption{option.New(...)}
}
```

Reply:
the snapshot of the current status is here: https://github.com/lestrrat-go/backoff/pull/25/commits/b5cd454a34a569b47545f4b74ae35280575399d1

Reply:
yeah, we are on the same page.